k9mail-library/src/test/java/com/fsck/k9/mail/store/imap/ImapFolderTest.java

```diff
@@ -1238,4 +1240,38 @@ private void prepareImapFolderForOpen(int openMode) throws MessagingException, I
     private void assertCheckOpenErrorMessage(String folderName, MessagingException e) {
         assertEquals("Folder " + folderName + " is not open.", e.getMessage());
     }
+
+    @Test
+    public void equals_returnsFalse_whenCaseDifferent() {
+        ImapFolder imapFolder = createFolder("Folder");
+        ImapFolder imapFolder2 = createFolder("folder");
+        assertNotEquals(imapFolder.hashCode(), imapFolder2.hashCode());
```

Comment:
Tests should only test one thing, so you know what went wrong from looking at the test name.

Also, there's no requirement for `hashCode()` to return different values for instances that are not equal. It's just a desired property. However, testing whether `hashCode()` contains a decent implementation is quite a bit of work. And not really necessary in this case since you simply delegate to `String.hashCode()`.